### ISSUE TITLE
R 3.4.0 update

### ISSUE THREAD

**Issue by roryk** [conda-forge/r-assertive.types-feedstock#3]
Last one, I hope: @conda-forge-admin, please rerender. Thank you!

**Comment by jdblischak** [comment]
@roryk @bgruening What is the plan here? This recipe was already built for R 3.4.1 with conda-build 2 (`r3.4.1`). This Issue and its subsequent PR #4 built the package with conda-build 3 (`r341`). I assume at some point we'll need to rerender all the R packages to be rebuilt with conda-build 3 so that the builds are consistently `r341`. I think we should do this in more coordinated fashion to save time (opening an Issue on every single feedstock would take a long time and be difficult to keep track of progress). Also, this mixing and matching of conda-build 2 and conda-build 3 builds confuses conda.

For example the following will install the `r341` builds of r-assertive and r-assertive.base, but the `r3.3.2` builds of all the other R packages (including r-base!):

<details>
<summary>
<code>
conda create --dry-run -n assert r-assertive
</code>
</summary>
<code>
Solving environment: done

## Package Plan ##

  environment location: /home/jdb-work/miniconda3-conda-forge/envs/assert

  added / updated specs: 
    - r-assertive


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    r-stringi-1.1.7            |         r3.3.2_1         697 KB  conda-forge
    r-assertive.reflection-0.0_4|         r3.3.2_0          97 KB  conda-forge
    r-codetools-0.2_15         |         r3.3.2_0          47 KB  conda-forge
    r-assertive.types-0.0_3    |           r341_0         450 KB  conda-forge
    r-assertive.base-0.0_7     |           r341_0         127 KB  conda-forge
    r-assertive.models-0.0_1   |         r3.3.2_0          29 KB  conda-forge
    r-assertive.numbers-0.0_2  |         r3.3.2_0          77 KB  conda-forge
    r-assertive.data-0.0_1     |         r3.3.2_0          54 KB  conda-forge
    r-assertive.properties-0.0_4|         r3.3.2_0          82 KB  conda-forge
    r-assertive.files-0.0_2    |         r3.3.2_0          75 KB  conda-forge
    r-assertive.strings-0.0_3  |         r3.3.2_0          63 KB  conda-forge
    r-assertive.matrices-0.0_1 |         r3.3.2_0          40 KB  conda-forge
    r-assertive.data.uk-0.0_1  |         r3.3.2_0          42 KB  conda-forge
    r-assertive.sets-0.0_3     |         r3.3.2_0          30 KB  conda-forge
    r-assertive.datetimes-0.0_2|           r341_0          35 KB  conda-forge
    r-assertive-0.3_5          |           r341_0         1.0 MB  conda-forge
    r-assertive.data.us-0.0_1  |         r3.3.2_0          36 KB  conda-forge
    r-assertive.code-0.0_1     |         r3.3.2_0          51 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         3.0 MB

The following NEW packages will be INSTALLED:

    bzip2:                  1.0.6-1         conda-forge
    ca-certificates:        2018.4.16-0     conda-forge
    cairo:                  1.14.6-4        conda-forge
    curl:                   7.54.1-0        conda-forge
    fontconfig:             2.12.1-6        conda-forge
    freetype:               2.7-1           conda-forge
    gettext:                0.19.8.1-0      conda-forge
    glib:                   2.51.4-0        conda-forge
    graphite2:              1.3.11-0        conda-forge
    gsl:                    2.1-2           conda-forge
    harfbuzz:               1.4.3-0         conda-forge
    icu:                    58.2-0          conda-forge
    jpeg:                   9b-2            conda-forge
    krb5:                   1.14.6-0        conda-forge
    libffi:                 3.2.1-3         conda-forge
    libgcc:                 5.2.0-0                    
    libiconv:               1.15-0          conda-forge
    libpng:                 1.6.28-1        conda-forge
    libssh2:                1.8.0-1         conda-forge
    libtiff:                4.0.7-0         conda-forge
    libxcb:                 1.13-0          conda-forge
    libxml2:                2.9.5-0         conda-forge
    ncurses:                5.9-10          conda-forge
    openssl:                1.0.2o-0        conda-forge
    pango:                  1.40.4-0        conda-forge
    pcre:                   8.41-1          conda-forge
    pixman:                 0.34.0-1        conda-forge
    r-assertive:            0.3_5-r341_0    conda-forge
    r-assertive.base:       0.0_7-r341_0    conda-forge
    r-assertive.code:       0.0_1-r3.3.2_0  conda-forge
    r-assertive.data:       0.0_1-r3.3.2_0  conda-forge
    r-assertive.data.uk:    0.0_1-r3.3.2_0  conda-forge
    r-assertive.data.us:    0.0_1-r3.3.2_0  conda-forge
    r-assertive.datetimes:  0.0_2-r341_0    conda-forge
    r-assertive.files:      0.0_2-r3.3.2_0  conda-forge
    r-assertive.matrices:   0.0_1-r3.3.2_0  conda-forge
    r-assertive.models:     0.0_1-r3.3.2_0  conda-forge
    r-assertive.numbers:    0.0_2-r3.3.2_0  conda-forge
    r-assertive.properties: 0.0_4-r3.3.2_0  conda-forge
    r-assertive.reflection: 0.0_4-r3.3.2_0  conda-forge
    r-assertive.sets:       0.0_3-r3.3.2_0  conda-forge
    r-assertive.strings:    0.0_3-r3.3.2_0  conda-forge
    r-assertive.types:      0.0_3-r341_0    conda-forge
    r-base:                 3.3.2-5         conda-forge
    r-codetools:            0.2_15-r3.3.2_0 conda-forge
    r-digest:               0.6.12-r3.3.2_0 bioconda   
    r-evaluate:             0.10.1-r3.3.2_0 conda-forge
    r-formatr:              1.5-r3.3.2_0    conda-forge
    r-glue:                 1.2.0-r3.3.2_0  conda-forge
    r-highr:                0.6-r3.3.2_0    conda-forge
    r-knitr:                1.12.3-r3.3.2_0 bioconda   
    r-magrittr:             1.5-r3.3.2_0    conda-forge
    r-markdown:             0.8-r3.3.2_1    conda-forge
    r-mime:                 0.5-r3.3.2_0    conda-forge
    r-stringi:              1.1.7-r3.3.2_1  conda-forge
    r-stringr:              1.3.0-r3.3.2_0  conda-forge
    r-yaml:                 2.1.13-r3.3.2_1 bioconda   
    readline:               6.2-0           conda-forge
    tk:                     8.5.19-2        conda-forge
    xorg-kbproto:           1.0.7-1         conda-forge
    xorg-libx11:            1.6.5-0         conda-forge
    xorg-libxau:            1.0.8-3         conda-forge
    xorg-libxdmcp:          1.1.2-3         conda-forge
    xorg-libxrender:        0.9.10-0        conda-forge
    xorg-renderproto:       0.11.1-1        conda-forge
    xorg-xproto:            7.0.31-6        conda-forge
    xz:                     5.2.3-0         conda-forge
    zlib:                   1.2.8-3         conda-forge

</code>
</details>
</br>

Pinning r-base to version `3.4.1` properly installs the `r3.4.1` versions of r-base and most of the other packages, but still `r341` of r-assertive and r-assertive.base:

<details>
<summary>
<code>
conda create --dry-run -n assert r-assertive r-base=3.4.1
</code>
</summary>
<code>
Solving environment: done

## Package Plan ##

  environment location: /home/jdb-work/miniconda3-conda-forge/envs/assert

  added / updated specs: 
    - r-assertive
    - r-base=3.4.1


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    r-assertive.numbers-0.0_2  |         r3.4.1_0          78 KB  conda-forge
    r-assertive.properties-0.0_4|         r3.4.1_0          83 KB  conda-forge
    r-assertive.data.us-0.0_1  |         r3.4.1_0          36 KB  conda-forge
    r-assertive.strings-0.0_3  |         r3.4.1_0          64 KB  conda-forge
    r-assertive.code-0.0_1     |         r3.4.1_0          52 KB  conda-forge
    r-assertive.reflection-0.0_4|         r3.4.1_0          97 KB  conda-forge
    r-assertive.sets-0.0_3     |         r3.4.1_0          30 KB  conda-forge
    r-assertive.matrices-0.0_1 |         r3.4.1_0          40 KB  conda-forge
    r-assertive-0.3_5          |           r341_0         1.0 MB  conda-forge
    r-assertive.datetimes-0.0_2|           r341_0          35 KB  conda-forge
    r-assertive.models-0.0_1   |         r3.4.1_0          29 KB  conda-forge
    r-stringi-1.1.7            |         r3.4.1_1         697 KB  conda-forge
    r-assertive.data.uk-0.0_1  |         r3.4.1_0          42 KB  conda-forge
    r-assertive.data-0.0_1     |         r3.4.1_0          54 KB  conda-forge
    r-assertive.files-0.0_2    |         r3.4.1_0          82 KB  conda-forge
    r-assertive.types-0.0_3    |           r341_0         450 KB  conda-forge
    r-assertive.base-0.0_7     |           r341_0         127 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         3.0 MB

The following NEW packages will be INSTALLED:

    bzip2:                  1.0.6-1         conda-forge
    ca-certificates:        2018.4.16-0     conda-forge
    cairo:                  1.14.10-0       conda-forge
    curl:                   7.59.0-1        conda-forge
    fontconfig:             2.12.6-0        conda-forge
    freetype:               2.8.1-0         conda-forge
    gettext:                0.19.8.1-0      conda-forge
    glib:                   2.55.0-0        conda-forge
    graphite2:              1.3.11-0        conda-forge
    gsl:                    2.1-2           conda-forge
    harfbuzz:               1.7.6-0         conda-forge
    icu:                    58.2-0          conda-forge
    jpeg:                   9b-2            conda-forge
    krb5:                   1.14.6-0        conda-forge
    libffi:                 3.2.1-3         conda-forge
    libgcc:                 5.2.0-0                    
    libiconv:               1.15-0          conda-forge
    libpng:                 1.6.34-0        conda-forge
    libssh2:                1.8.0-2         conda-forge
    libtiff:                4.0.9-0         conda-forge
    libxcb:                 1.13-0          conda-forge
    libxml2:                2.9.8-0         conda-forge
    ncurses:                5.9-10          conda-forge
    openssl:                1.0.2o-0        conda-forge
    pango:                  1.40.14-0       conda-forge
    pcre:                   8.39-0          conda-forge
    pixman:                 0.34.0-1        conda-forge
    r-assertive:            0.3_5-r341_0    conda-forge
    r-assertive.base:       0.0_7-r341_0    conda-forge
    r-assertive.code:       0.0_1-r3.4.1_0  conda-forge
    r-assertive.data:       0.0_1-r3.4.1_0  conda-forge
    r-assertive.data.uk:    0.0_1-r3.4.1_0  conda-forge
    r-assertive.data.us:    0.0_1-r3.4.1_0  conda-forge
    r-assertive.datetimes:  0.0_2-r341_0    conda-forge
    r-assertive.files:      0.0_2-r3.4.1_0  conda-forge
    r-assertive.matrices:   0.0_1-r3.4.1_0  conda-forge
    r-assertive.models:     0.0_1-r3.4.1_0  conda-forge
    r-assertive.numbers:    0.0_2-r3.4.1_0  conda-forge
    r-assertive.properties: 0.0_4-r3.4.1_0  conda-forge
    r-assertive.reflection: 0.0_4-r3.4.1_0  conda-forge
    r-assertive.sets:       0.0_3-r3.4.1_0  conda-forge
    r-assertive.strings:    0.0_3-r3.4.1_0  conda-forge
    r-assertive.types:      0.0_3-r341_0    conda-forge
    r-base:                 3.4.1-3         conda-forge
    r-codetools:            0.2_15-r3.4.1_0 conda-forge
    r-digest:               0.6.12-r3.4.1_0 conda-forge
    r-evaluate:             0.10.1-r3.4.1_0 conda-forge
    r-glue:                 1.2.0-r3.4.1_0  conda-forge
    r-highr:                0.6-r3.4.1_0    conda-forge
    r-knitr:                1.20-r3.4.1_0   conda-forge
    r-magrittr:             1.5-r3.4.1_0    conda-forge
    r-markdown:             0.8-r3.4.1_1    conda-forge
    r-mime:                 0.5-r3.4.1_0    conda-forge
    r-stringi:              1.1.7-r3.4.1_1  conda-forge
    r-stringr:              1.3.0-r3.4.1_0  conda-forge
    r-yaml:                 2.1.18-r3.4.1_0 conda-forge
    readline:               7.0-0           conda-forge
    tk:                     8.6.7-0         conda-forge
    xorg-kbproto:           1.0.7-1         conda-forge
    xorg-libx11:            1.6.5-0         conda-forge
    xorg-libxau:            1.0.8-3         conda-forge
    xorg-libxdmcp:          1.1.2-3         conda-forge
    xorg-libxrender:        0.9.10-0        conda-forge
    xorg-renderproto:       0.11.1-1        conda-forge
    xorg-xproto:            7.0.31-6        conda-forge
    xz:                     5.2.3-0         conda-forge
    zlib:                   1.2.11-0        conda-forge

</code>
</details>

**Comment by jdblischak** [comment]
Actually, this takes a few more steps to reproduce exactly. The main issue is that to conda, `r341` != `r3.4.1`. However, this problem usually gets masked since R 3.4.1 gets installed by default anyways. Thus the commands I shared above appear to work just fine when run in a standard conda-forge setup with the channels conda-forge and defaults (I tested this in the Docker image condaforge/linux-anvil).

Here are multiple ways to demonstrate that `r341` != `r3.4.1`:

* Use the bioconda channel:

```
conda create --dry-run -n assert r-assertive -c bioconda -c conda-forge -c defaults --override-channels
```

This will install R 3.3.2 for almost everything except for the r-assertive packages with a `r341`build (the fact that bioconda R packages force conda to downgrade to R 3.3.2 is a separate can of worms).

* Pin r-base to 3.3.2 but not r-assertive

```
conda create --dry-run -n assert r-assertive r-base==3.3.2 -c conda-forge -c defaults --override-channels
```

Once again, all the R packages are `r3.3.2` except for the r-assertive packages with `r341` builds. These packages have `r3.3.2` builds available that should be installed, but once the build names are changed to the conda-build 3 version, they are given priority.
